src/ExtractTileOperations.cpp

```diff
@@ -143,6 +164,126 @@ Tile<3> get_3d_tile_index(const Expr &e) {
     return {true, base, {x_stride, 0, r_stride}, {x_tile, y_tile, r_tile}};
 }
 
+Tile<3> get_3d_rhs_tile_index(const Expr &e, int element_width) {
+    const auto *sub = e.as<Sub>();
+    const Add *add_lhs = nullptr;
+
+    // there's not always a sub pattern
+    if (!sub) {
+        add_lhs = e.as<Add>();
+    } else {
+        add_lhs = sub->a.as<Add>();
+    }
+
+    if (!add_lhs) {
+        return {};
+    }
+
+    // obtain the x, y, r dimensions
+    const Add *dim_expr = add_lhs->b.as<Add>();
+
+    const Broadcast *y_bc = dim_expr->b.as<Broadcast>();
+
+    int tile_y = y_bc->lanes;
+
+    const Mod *mod = dim_expr->a.as<Mod>();
+
+    const Broadcast *bc_ramp = mod->a.as<Broadcast>();
+
+    if (!bc_ramp) {
+        return {};
+    }
+
+    int tile_xy = bc_ramp->lanes;
+    int tile_x = tile_xy / tile_y;
+
+    const Ramp *r_ramp = bc_ramp->value.as<Ramp>();
+
+    if (!r_ramp) {
+        return {};
+    }
+
+    int tile_r = r_ramp->lanes;
+
+    // get the base and stride
+    const Broadcast *base_stride_bc = add_lhs->b.as<Add>()->b.as<Broadcast>();
```

Comment:
This should be:
```
const Broadcast *base_stride_bc = dim_expr->b.as<Broadcast>();
```

Reply:
Actually, it seems that `base_stride_bc` is the exact same as `y_bc`. Seems to be a bit of duplicated logic here, should one of these series of checks be checking the `a` side of `add_lhs`?

Reply:
resolved with 20b0c3aec52bd776a68bd5f9631d21c3b7685166